dataprofiler/data_readers/data_utils.py

```diff
@@ -301,7 +300,46 @@ def detect_file_encoding(file_path, buffer_size=1024, max_lines=20):
     encoding = detector.result["encoding"]
 
     # Typical file representation is utf-8 instead of ascii, treat as such.
-    if not encoding or encoding == 'ascii':
+    if not encoding or encoding == 'ascii' or encoding == 'Windows-1254':
+        encoding = 'utf-8'
+
+    # Check if encoding can be used to decode without throwing an error
+    def _decode_is_valid(encoding):
+        try: 
+            open(file_path, "rb").read().decode(encoding.lower())
```

Comment:
If you're going to use this, you should limit the number of bytes (could be 1000Gb). Recommend no more than 1Mb

Reply:
I read in 1Mb = 1024*1024 bytes rather the entire file and switched to using encoding=encoding in the open command in the new commit